src/App.tsx

```diff
@@ -1,117 +1,127 @@
+            return value
+        });
+        return newState;
+
+    } else if (action.type === 'remove') {
+        console.log("remove", action.dataId);
+        const newState: PowerUsers[] = state.filter(function (item) {
+            return item.id !== action.dataId
+        });
+        return newState
+
+    } else {
+        throw new Error(`This action type isn't supported.`)
+    }
 }
 
 
 const App: React.FC = () => {
 
-  
-
-  const [UserService] = useState(new UsersService())
-  const [searchField, setSearchField] = useState("");
-  const [users, dispatch] = React.useReducer(
-    reducer,
-    []
-  )
-
-
-  function handleSearchChange (event: React.ChangeEvent<HTMLInputElement>) {
-    setSearchField(event.target.value);
-  }
-
-  function handleUserUpdate (id:number) {
-    dispatch({
-      'type': "update",
-      'dataUsers': [],
-      'dataId':id
-    })
-  }
-
-  function handleUserRemove (id:number) {
-    dispatch({
-      'type': "remove",
-      'dataUsers': [],
-      'dataId':id
-    })
-    UserService.removeUser(id);
-  }
-
-  React.useEffect(() => {
-    async function setAllUsers() {
-      let powerUsers :PowerUsers[] = await UserService.getAllUsers();
-      dispatch({
-        'type': "initialize",
-        'dataUsers': powerUsers,
-        'dataId':0
-      })
+
+    const [searchField, setSearchField] = useState("");
+    const [users, dispatch] = React.useReducer(
+        reducer,
+        []
+    );
+
+
+    function handleSearchChange(event: React.ChangeEvent<HTMLInputElement>) {
+        setSearchField(event.target.value);
     }
-    setAllUsers();
-  }, [UserService]);
 
-  React.useEffect(() => {
-    const id = window.setTimeout(() => {
-      async function setSearchedUsers() {
-        let powerUsers :PowerUsers[] = await UserService.getUser(searchField);
+    function handleUserUpdate(id: number) {
         dispatch({
-          'type': "filter",
-          'dataUsers': powerUsers,
-          'dataId':0
+            'type': "update",
+            'dataUsers': [],
+            'dataId': id
         })
-      }
-      setSearchedUsers();
-    }, 300)
-    return () => {
-      window.clearTimeout(id)
     }
-  }, [searchField,UserService])
-
-
-  return (
-    <div >
-      <SearchBar handleSearchChange = {handleSearchChange}></SearchBar>
-      <UsersList powerUsers={users}
-                 handleUserUpdate = {handleUserUpdate}
-                 handleUserRemove = {handleUserRemove}
-      />
-    </div>
-  );
+
+    function handleUserRemove(id: number) {
+        dispatch({
+            'type': "remove",
+            'dataUsers': [],
+            'dataId': id
+        });
+        UsersService.removeUser(id);
+    }
+
+    React.useEffect(() => {
+        async function setAllUsers() {
+            const powerUsers = await UsersService.getAllUsers();
+            dispatch({
+                type: "initialize",
+                dataUsers: powerUsers.map(pUser => ({...pUser, count: 0})),
```

Comment:
Here is is required to provide count if what is returned from `UsersService.getAllUsers()` does not have it.

I have tried to make more TS friendly the UsersService module, and here it was catching the error about missing count :)